tests/rpc-tests/rpc-test.js

```diff
@@ -0,0 +1,777 @@
+        sendTestData[i].req.method === 'eth_sendTransaction')
+    ) {
+      continue
+    }
+
+    if (sendTestData[i].req.method === 'eth_sign') {
+      sendTestData[i].req.params[0] = sender
+    } else if (
+      sendTestData[i].req.method === 'eth_sendTransaction' ||
+      sendTestData[i].req.method === 'eth_signTransaction'
+    ) {
+      sendTestData[i].req.params[0].from = sender
+    }
+
+    // raw transaction to be used in eth_sendRawTransaction
+    if (sendTestData[i].req.method === 'eth_signTransaction') {
+      sendTestData[i].req.params[0].nonce = web3.utils.toHex(nonce)
+      response = await axiosInstance.post('/', sendTestData[i].req)
+      console.log(`signed tx: ${response.data.result.raw}`)
+    }
+  }
+
+  for (let i = 0; i < sendTestData.length; i++) {
+    if (process.env.RPC_URL) {
+      continue
+    }
+    if (sendTestData[i].req.method === 'eth_sendRawTransaction') {
+      sendTestData[i].req.params[0] = response.data.result.raw
+      break
+    }
+  }
+}
+
+function sortAccessList(accessList) {
+  const sortedAccessList = _.sortBy(accessList, 'address')
+
+  _.forEach(sortedAccessList, (obj) => {
+    obj.storageKeys = _.sortBy(obj.storageKeys)
+  })
+
+  return sortedAccessList
+}
+
+export async function rpcTest() {
+  try {
+    const [machine, web3, sender] = await init()
+
+    const getCurrentSnapshot = {
+      method: 'bor_getSnapshot',
+      params: ['latest'],
+      id: 1,
+      jsonrpc: '2.0'
+    }
+
+    // Getter rpc calls
+    let axiosInstance
+    const getTestData = fetchTestFiles(
+      `../../tests/rpc-tests/RPC-testdata/${process.env.RPC_NETWORK}/getters`
+    )
+    if (process.env.RPC_URL) {
+      axiosInstance = axios.create({
+        baseURL: `${machine}`
+      })
+    } else {
+      axiosInstance = axios.create({
+        baseURL: `http://${machine}:8545`
+      })
+    }
+
+    let response
+
+    let getterIterattions = process.env.EXECUTION_COUNT_GETTERS
+    if (
+      getterIterattions <= 0 ||
+      getterIterattions === null ||
+      getterIterattions === undefined
+    ) {
+      getterIterattions = 1
+    }
+
+    for (let iter = 0; iter < getterIterattions; iter++) {
+      console.log('📍Executing getter rpc calls. Iteration: ', iter)
+      for (let i = 0; i < getTestData.length; i++) {
+        console.log('📍 Executing: ', getTestData[i].req.method)
+        response = await axiosInstance.post('/', getTestData[i].req)
+
+        if (getTestData[i].req.method === 'bor_getCurrentProposer') {
+          const currentSnapshotResponse = await axiosInstance.post(
+            '/',
+            getCurrentSnapshot
+          )
+          // If the validator set changes, the assertion will fail
+          response = await axiosInstance.post('/', getTestData[i].req)
+          assert.deepStrictEqual(
+            currentSnapshotResponse.data.result.validatorSet.proposer.signer,
+            response.data.result
+          )
+          continue
```

Comment:
all the methods that requires validators changes are marked with a `continue` now
imho,  this can be embedded in the test itself. Something like `isResponseDynamic=true` will only run once